frame/contracts/primitives/src/lib.rs

```diff
@@ -71,15 +71,18 @@ pub struct ContractResult<R, Balance> {
 	pub debug_message: Vec<u8>,
 	/// The execution result of the wasm code.
 	pub result: R,
+	/// The events that were emitted during execution. During on-chain execution these events are
+	/// None to avoid putting all the Events emitted in the block so far into the PoV.
+	pub events: Option<Vec<EventRecord>>,
```

Comment:
We should not mention PoV here. I reads as if the inclusion into this struct would put them into the PoV. I think it is enough to say that it is an option because event collection is optional.